packages/rest/src/rest.application.ts

```diff
@@ -65,11 +72,29 @@ export class RestApplication extends Application implements HttpServerLike {
     return this.restServer.requestHandler;
   }
 
-  constructor(config: ApplicationConfig = {}) {
+  constructor(readonly config: ApplicationConfig = {}) {
     super(config);
     this.component(RestComponent);
   }
 
+  component(componentCtor: Constructor<Component>, name?: string) {
```

Comment:
What if an application does not extend from `RestApplication` but mount the `RestComponent`?

Reply:
If the app is extending from `Application`, then there is no REST server configured by default and the rest-specific exports are silently ignored.

Reply:
What I'm trying to say is that `RestComponent` should be responsible for mounting such routes. In this case, other components contribute routes for the `RestComponent`.

Reply:
@raymondfeng can you elaborate on the above statement, please?

Reply:
A developer can do the the following:

```ts
const app = new Application();
app.component(RestComponent);
app.component(ExplorerComponent);
```
With the current implementation, no `/explorer` route will be mounted.

Reply:
Thanks, that explains it.

@bajtos that would require us to leave`RestApplication.component` alone and let `Application.component` take care of mounting components. However, now we'll have to have a list of supported components (`Component | RestComponent` etc.) in the parameters, IF we don't want `RestComponent` to extend `Component`.

Reply:
```ts
const app = new Application();
app.component(RestComponent);
app.component(ExplorerComponent);
```

Good catch! I did not realize we have `RestComponent`.

I need to think this through, will post a follow-up comment later.

Reply:
As I commented at the bottom, I think this is a wider problem that we should address outside of this pull request, I opened a new issue for that - see https://github.com/strongloop/loopback-next/issues/1967